Sources/NIOCore/EventLoop.swift

```diff
@@ -356,6 +356,14 @@ public protocol EventLoop: EventLoopGroup {
     /// It is valid for an `EventLoop` not to implement any of the two `_promise` functions. If either of them are implemented,
     /// however, both of them should be implemented.
     func _promiseCompleted(futureIdentifier: _NIOEventLoopFutureIdentifier) -> (file: StaticString, line: UInt)?
+
+    /// Set a timer that will call a handler at the given time.
+    @discardableResult
+    func setTimer(for deadline: NIODeadline, _ handler: any NIOTimerHandler) -> NIOTimer
```

Comment:
I'm not sure about hiding the label of the `handler` here, I think the API would be clearer if users have to explicitly set `handler` (`_ handler: ...` is typically a closure)

Reply:
NIT: I am wondering if we should retain the `schedule` naming here and make this `scheduleTimer` instead of `setTimer`.

Reply:
I don't think you _schedule_ a timer because it can imply that the timer doesn't start immediately.

If it's more important to keep the _schedule_ verb, then we can have something like `scheduleCallback`, and if you really want to mix the _schedule_ verb with the `_timer_` noun, it would need to be something like `scheduleTimerToFinish(in:)` and `scheduleTimerToFinish(at:)`.

Reply:
FWIW I don't like the "set" here, it sounds like we're providing a timer and storing it on the event loop, rather than having the event loop create a timer for us.

I'm okay with `scheduleCallback` – I think that actually best describes what the function does.

Reply:
OK, the summary of this thread is two fold:
1. We want to have the `handler` label (i.e. no underscore).
2. We didn't like the _set_ and/or _timer_ language in the API.

The relevant targeted changes for these pieces of feedback are contained in their own commits:
- (1) is addressed by 950db0c84.
- (2) is addressed by ceabc7b28.